hydrus/helpers.py

```diff
@@ -530,3 +530,21 @@ def parse_collection_members(object_: dict) -> dict:
             return error_response(error)
     object_['members'] = members
     return object_
+
+
+def get_fragments(resource: str) -> dict:
+    """Gets a fragment of the main hydra vocabulary.
+
+    :param resource: Fragment specified in the request parameters
+    :type resource: str
+    :return: Object referred to by the fragment
+    :rtype: dict
+    """
+    resourcedict = dict()
+    gen_doc = get_doc().generate()
+    resourcedict['@context'] = {i: gen_doc['@context'][i] for i in sorted(gen_doc['@context'])}
+    vocab_route = get_doc().doc_name
+    matching_string = f'{get_hydrus_server_url()}{get_api_name()}/{vocab_route}'+'#'+resource
+    res = next((i for i in gen_doc['supportedClass'] if i['@id'] == matching_string), None)
```

Comment:
What exactly are you trying to here? (I think this line could be rewritten in a more readable way, instead of having a compact one-liner)

Reply:
So the line 548, is for basically iterating through all of the id's in the apidoc and finding the resource that is specified. Since the id's have a fragment identifier in them, in line 547, I just appended the resource specified with the vocab URI and a # before the resource. 

The reason I used the next keyword and converted it into a compact one liner is because I have read that iterating using next is faster usually. Actually this can be converted into a simple for loop block, if that is more suited I will change the line.

Reply:
Okay, I see. I would recommend converting this into a simple for loop block for readability and maintainability purposes.